Sage/carbon
# docs(accordion): add documentation about wrapping component FE-4702

### Proposed behaviour


Refactored: 
* `Carousel`: Moves the styling that handles the visible slide into the `Carousel` styles and removes unneeded use
of `cloneElement`
* `Checkbox`: Replaces iteration over `Checkbox` children and use of `cloneElement` to apply validation interface with
`context` instead
* `Dialog` and `Form`: remove `cloneElement` and `dialogRef`
* `InlineInputs`: To support being able to wrap the children inputs the use of `cloneElement` in `InlineInputs`
component has been removed in favour of using context. The `InputBehaviour` component  has been
updated to consume the new `InlineInputsContext` and pass the `aria-labelledby` value into the
`Input` and `HiddenCheckableInput` components.
* `StepSequence`: replace use of `cloneElement` with `context`

Updated docs on wrapping children for `Accordion`, `RadioButton`, `ActionPopover`, `Card`, `Pages`, `DuellingPickList`, `FlatTableBodyDraggable`, `MultiActionButton`, `SplitButton`, `Note`, `TextEditor`, `Select`s, `SimpleColorPicker`, `Tile`, `TileSelect` and `Menu`.

### Current behaviour


`Carousel`, `Checkbox`, `Dialog`, `InlineInputs`, `StepSequence` use cloning instead of context

No docs about wrapping `Accordion`, `RadioButton`, `ActionPopover`, `Card`, `Pages`, `DuellingPickList`, `FlatTableBodyDraggable`, `MultiActionButton`, `SplitButton`, `Note`, `TextEditor`, `Select`s, `SimpleColorPicker`, `Tile`, `TileSelect` and `Menu`.

### Checklist




#### QA


### Additional context



### Testing instructions


`Carousel`, `Checkbox`, `Dialog`, `InlineInputs`, `StepSequence` stories will need testing to ensure no regressions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
